ipn/ipnlocal/local_test.go

```diff
@@ -1330,3 +1331,281 @@ func (rc *routeCollector) AdvertiseRoute(pfx netip.Prefix) error {
 	rc.routes = append(rc.routes, pfx)
 	return nil
 }
+
+type mockSyspolicyHandler struct {
+	t             *testing.T
+	key           syspolicy.Key
+	s             string
+	exitNodeIDKey bool
+	exitNodeIPKey bool
+	exitNodeID    string
+	exitNodeIP    string
+	bothKeys      bool
+	err           error
+}
+
+func (h *mockSyspolicyHandler) ReadString(key string) (string, error) {
+	if h.exitNodeIDKey && key == string(syspolicy.ExitNodeID) {
+		return h.exitNodeID, h.err
+	}
+	if h.exitNodeIPKey && key == string(syspolicy.ExitNodeIP) {
+		return h.exitNodeIP, h.err
+	}
+	return "", h.err
+}
+
+func (h *mockSyspolicyHandler) ReadUInt64(key string) (uint64, error) {
+	return 0, h.err
+}
+
+func (h *mockSyspolicyHandler) ReadBoolean(key string) (bool, error) {
+	return false, h.err
+}
```

Comment:
I don't think `h.err` is ever being used, so let's trim it down to just the things that get used.

```suggestion
type mockSyspolicyHandler struct {
	t             *testing.T
	key           syspolicy.Key
	s             string
	exitNodeIDKey bool
	exitNodeIPKey bool
	exitNodeID    string
	exitNodeIP    string
}

func (h *mockSyspolicyHandler) ReadString(key string) (string, error) {
	if h.exitNodeIDKey && key == string(syspolicy.ExitNodeID) {
		return h.exitNodeID, nil
	}
	if h.exitNodeIPKey && key == string(syspolicy.ExitNodeIP) {
		return h.exitNodeIP, nil
	}
	return "", syspolicy.ErrNoSuchKey
}

func (h *mockSyspolicyHandler) ReadUInt64(key string) (uint64, error) {
	return 0, syspolicy.ErrNoSuchKey
}

func (h *mockSyspolicyHandler) ReadBoolean(key string) (bool, error) {
	return false, syspolicy.ErrNoSuchKey
}
```